src/Functions/generateRandomStructure.cpp

```diff
@@ -0,0 +1,379 @@
+                writeLowCardinalityNestedType(rng, buf);
+                writeChar(')', buf);
+                return;
+            case TypeIndex::Nullable:
+            {
+                writeCString("Nullable(", buf);
+                writeRandomType<false>(column_name, rng, buf, depth + 1);
+                writeChar(')', buf);
+                return;
+            }
+            case TypeIndex::Array:
+            {
+                writeCString("Array(", buf);
+                writeRandomType(column_name, rng, buf, depth + 1);
+                writeChar(')', buf);
+                return;
+            }
+            case TypeIndex::Map:
+            {
+                writeCString("Map(", buf);
+                writeMapKeyType(rng, buf);
+                writeCString(", ", buf);
+                writeRandomType(column_name, rng, buf, depth + 1);
+                writeChar(')', buf);
+                return;
+            }
+            case TypeIndex::Tuple:
+            {
+                size_t elements = rng() % MAX_TUPLE_ELEMENTS + 1;
+                bool generate_nested = rng() % 2;
+                bool generate_named_tuple = rng() % 2;
+                if (generate_nested)
+                    writeCString("Nested(", buf);
+                else
+                    writeCString("Tuple(", buf);
+
+                for (size_t i = 0; i != elements; ++i)
+                {
+                    if (i != 0)
+                        writeCString(", ", buf);
+
+                    String element_name = "e" + std::to_string(i + 1);
+                    if (generate_named_tuple || generate_nested)
+                    {
+                        writeString(element_name, buf);
+                        writeChar(' ', buf);
+                    }
+                    writeRandomType(element_name, rng, buf, depth + 1);
+                }
+                writeChar(')', buf);
+                return;
+            }
+            default:
+                writeString(magic_enum::enum_name<TypeIndex>(type), buf);
+                return;
+        }
+    }
+
+    void writeMapKeyType(pcg64 & rng, WriteBuffer & buf) const
+    {
+        TypeIndex type = map_key_types[rng() % map_key_types.size()];
+        if (type == TypeIndex::FixedString)
+            writeString("FixedString(" + std::to_string(rng() % MAX_FIXEDSTRING_SIZE_WITHOUT_SUSPICIOUS + 1) + ")", buf);
+        else
+            writeString(magic_enum::enum_name<TypeIndex>(type), buf);
+    }
+
+    void writeLowCardinalityNestedType(pcg64 & rng, WriteBuffer & buf) const
+    {
+        /// Support only String and FixedString (maybe Nullable).
+        String nested_type;
+        bool make_nullable = rng() % 2;
+        if (make_nullable)
+            writeCString("Nullable(", buf);
+
+        if (rng() % 2)
+            writeCString("String", buf);
+        else
+            writeString("FixedString(" + std::to_string(rng() % MAX_FIXEDSTRING_SIZE_WITHOUT_SUSPICIOUS + 1) + ")", buf);
```

Comment:
For completeness, could check setting `allow_suspicious_low_cardinality_types` and allow all other supported types if true. The supported types appear to be map_key_types + Float32, Float64, and IPv6.

Reply:
Agree, done